superset/utils/async_query_manager.py

```diff
@@ -115,7 +117,13 @@ def init_app(self, app: Flask) -> None:
         def validate_session(  # pylint: disable=unused-variable
             response: Response,
         ) -> Response:
-            user_id = session["user_id"] if "user_id" in session else None
+            user_id = None
+
+            try:
+                user_id = g.user.get_id()
+                user_id = int(user_id)
```

Comment:
Casting to an int isn't going to work if `user_id` is `None` here.

Reply:
Right, I changed the `except` statement to catch `exception` for this reason (and to be extra careful, given that this is run for all requests). I can switch to conditions if you'd prefer.

Reply:
I don't have strong feelings, but it may be more clear with an explicit condition.